src/app/widgets/Visualizer/Visualizer.jsx

```diff
@@ -1401,6 +1447,91 @@ class Visualizer extends Component {
         } else {
             setVisualizerReady();
         }
+
+        this.fileLoaded = true;
+    }
+
+    calculateLimits(data) {
+        /* get machine 0
+                0 is top right
+                1 is top left
+                2 bottom right
+                3 bottom left
+        */
+        let machineCorner = reduxStore.getState().controller.settings.settings.$23;
+
+        let xMultiplier = 1;
+        let yMultiplier = 1;
+        switch (machineCorner) {
+        case 0:
+            xMultiplier = -1;
+            yMultiplier = -1;
+            break;
+        case 1:
+            xMultiplier = 1;
+            yMultiplier = -1;
+            break;
+        case 2:
+            xMultiplier = -1;
+            yMultiplier = 1;
+            break;
+        case 3:
+        default:
+            xMultiplier = 1;
+            yMultiplier = 1;
+            break;
+        }
+
+        // get soft limits
+        let xMax = parseFloat(reduxStore.getState().controller.settings.settings.$130);
```

Comment:
I'd consider mapping the controller.settings to props since they a) don't change very often and b) you can replace 4 getState calls with the single map